deploy/helm/kubecf/job_moving.bzl

```diff
@@ -0,0 +1,39 @@
+"""
+An extension for extracting jobs from existing instance groups and creating new ones based on these
+job definitions.
+"""
+
+load("//rules/yaml_extractor:defs.bzl", "yaml_extractor")
+
+# The default list of jobs related to log-cache.
+LOG_CACHE_JOBS = [
+    "log-cache",
+    "log-cache-gateway",
+    "log-cache-nozzle",
+    "log-cache-cf-auth-proxy",
+    "route_registrar",
+]
+
+def extract_log_cache(instance_group, log_cache_jobs):
```

Comment:
Can we not use the existing `yaml_extractor` for this? Split it into 5 jobs files, and the re-combine them in `move_log_cache.yaml`?

I rather strongly prefer having fewer `*.bzl` files, as every time we drop into one it feels like we're working against bazel.

Reply:
Could you expand on `feels like we're working against bazel`?

Reply:
I see bazel as having two languages (that are actually the same starlark): the one in `*.bazel` where we define rules, and the one in `*.bzl` where we define implementation.  The two types of files feel very different and _feel_ like completely different things with a large mental context switch between them.  Sort of like dropping into inline assembly from within a C program.

As such, every time I need to dig into an implementation file I feel like I stopped writing starlark-the-rules-language and are dipping into starlark-the-build-system-implementation-language, and it feels like I'm working _around_ the existing bazel rules to do something it doesn't want me to do.

Reply:
I can see that. These python functions on `.bzl` files are known as macros in Bazel. They are sometimes useful for creating a bunch of targets. I was able to do the same job with a list comprehension on the `.build` file. It should be better to read now.

Reply:
Fixed in https://github.com/cloudfoundry-incubator/kubecf/pull/619/commits/52ebe7224e48656c3986a2d5d109f9083a56d106.